File: Math/floor_sqrt/C++/sqrt.cpp
// A C++ program to find floor(sqrt(x) using binary search
#include<bits/stdc++.h>
using namespace std;

// Returns floor of square root of x
int floorSqrt(int x)
{
    // Base cases
    if (x == 0 || x == 1)
    return x;

    // Staring from 1, try all numbers until
    // i*i is greater than or equal to x.
    int i = 1, result = 1;
    while (result <= x)
    {
      i++;
      result = i * i;
    }
    return i - 1;
}

int main()
{
	int x;
	cout<<"Enter a number"<<endl;
	cin>>x;
	cout << "The floor square root of an integer is "<< floorSqrt(x) << endl;
	return 0;
}

File: Math/floor_sqrt/sqrt.py
# Returns floor of square root of x
def sqrt(num) :
    if num == 0 or num == 1:
        return num
    start = 1
    end = num
    while start < end:
        mid = (start + end) // 2
        if (mid*mid == num) :
            return mid
        # Since we need floor, we update
        # answer when mid*mid is smaller
        # than x, and move closer to sqrt(x)
        if mid * mid < num:
            start = mid + 1
            ans = mid
        else :
            # If mid*mid is greater than x
            end = mid
    return ans

if __name__ == "__main__":
    print(sqrt(11)) #output is 3
    print(sqrt(4))  #output is 2
